pkg/hvs/controllers/flavortemplate_controller_test.go

```diff
@@ -0,0 +1,434 @@
+						"//host_info/suefi_enabled='true'"
+					],
+					"flavor_parts": {
+						"PLATFORM": {
+							"meta": {
+								"tpm_version": "2.0",
+								"uefi_enabled": true,
+								"vendor": "Linux"
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 0,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_equals": {}
+								}
+							]
+						},
+						"OS": {
+							"meta": {
+								"tpm_version": "2.0",
+								"uefi_enabled": true,
+								"vendor": "Linux"
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 7,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true,
+									"eventlog_includes": [
+										"shim",
+										"db",
+										"kek",
+										"vmlinuz"
+									]
+								}
+							]
+						}
+					}
+				}`
+
+				req, err := http.NewRequest(
+					"POST",
+					"/flavor-template",
+					strings.NewReader(flavorTemplateJson),
+				)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				req.Header.Set("Content-Type", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusCreated))
+			})
+		})
+
+		Context("Provide a FlavorTemplate data that contains invalid fileds, to validate against schema", func() {
+			It("Should get HTTP Status: 400", func() {
+				router.Handle("/flavor-template", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Create))).Methods("POST")
+				flavorgroupJson := `{
+					"label": "",
+					"condition": [
+						"//host_info/vendor='Linux'",
+						"//host_info/tpm_version='2.0'",
+						"//host_info/uefi_enabled='true'",
+						"//host_info/suefi_enabled='true'"
+					],
+					"flavor_parts": {
+						"PLATFORM": {
+							"meta": {
+								"tpm_version": "2.0",
+								"tboot_installed": true
+							},
+							"pcr_rules": [
+								{
+									"pcr": {
+										"index": 0,
+										"bank": "SHA256"
+									},
+									"pcr_matches": true
+								}
+							]
+						}
+					}
+				}`
+
+				req, err := http.NewRequest(
+					"POST",
+					"/flavor-template",
+					strings.NewReader(flavorgroupJson),
+				)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				req.Header.Set("Content-Type", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(400))
+			})
+		})
+
+		Context("Provide a empty data that should give bad request error", func() {
+			It("Should get HTTP Status: 400", func() {
+				router.Handle("/flavor-template", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Create))).Methods("POST")
+				req, err := http.NewRequest(
+					"POST",
+					"/flavor-template",
+					strings.NewReader(""),
+				)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
```

Comment:
Expect(err).NotTo(HaveOccurred()) must be before processing any req details

Reply:
Addressed.